### PR TITLE
bug-fix: when other user leaves, do not remove channel

### PR DESCRIPTION
**Pull request by aldenquimby**
- fixes a bug that may have existed for a while, or possibly have been caused by #901

### steps to reproduce:
1. User A creates a new group channel with 3+ users
2. User B leaves the channel

### expected:
- User A still sees the channel, with it's message history
- members are updated, and possibly channel avatar is updated too

### actual:
- channel immediately disappears for User A
- User A must refresh / reload the page to see the channel
- this is particularly jarring if it was your current channel and you were actively chatting

## External Contributions
- I know this PR cannot be merged directly
- I'm hoping that similar to #893 and #894 , someone on your team can pick it up

### CODE REVIEW

File: src/modules/ChannelList/dux/reducers.ts [hunk 1]
```diff
@@ -108,28 +108,36 @@ export default function channelListReducer(
       .with({ type: channelListActions.ON_USER_LEFT }, (action) => {
         const { channel, isMe } = action.payload;
         const { allChannels, currentUserId, currentChannel, channelListQuery, disableAutoSelect } = state;
-        let nextChannels = allChannels.filter((ch) => ch.url !== channel.url);
```
Comment (aldenquimby) [inline src/modules/ChannelList/dux/reducers.ts:111]: The diff here is worse than I'd like, and could be reduced if we want, but I found it much easier to reason about what was happening with an explicit top-level `isMe` if/else

File: src/modules/ChannelList/dux/reducers.ts [hunk 2]
```diff
@@ -108,28 +108,36 @@ export default function channelListReducer(
...
+          return {
+            ...state,
+            currentChannel: nextChannel,
+            allChannels: nextChannels,
+          };
+        }
+        // If someone else left: update allChannels if needed (don't touch currentChannel)
+        else {
+          let nextChannels = allChannels.filter((ch) => ch.url !== channel.url);
+          if (!channelListQuery || filterChannelListParams(channelListQuery, channel, currentUserId)) {
```
Comment (aldenquimby) [inline src/modules/ChannelList/dux/reducers.ts:131]: There's a change in logic here... if no channelListQuery, we should not be removing the channel (but the prior logic was incorrectly removing the channel)

**Comment by liamcho** [comment]
@aldenquimby Hi thanks for providing the bug fix PR. We are onto [fixing this bug](https://github.com/sendbird/sendbird-uikit-react/pull/947) and targeted to release in the next patch version, v3.10.2.

File: src/modules/ChannelList/dux/reducers.ts (hunk 2 above)
Comment (liamcho) [reply src/modules/ChannelList/dux/reducers.ts:130]: Yes you are correct, thanks for spotting this out.

File: src/ui/ChannelAvatar/index.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ function ChannelAvatar({
           alt={channel?.name}
         />
       )
-  ), [channel?.members, channel?.coverUrl, theme]);
+  ), [utils.getChannelAvatarSource(channel, userId), theme]);
```
[the remark this answers is not in the document]
Comment (liamcho) [reply src/ui/ChannelAvatar/index.tsx:69]: Yes correct! thanks for spotting this as well! React does not recognize changes of instance properties. We will apply your sugestion as well!

**Comment by liamcho** [comment]
Closing this PR because the [fix PR has been merged](https://github.com/sendbird/sendbird-uikit-react/pull/947)

**Comment by aldenquimby** [comment]
Thank you @liamcho ! Appreciate it 💪

**Pull request closed without merging**